projects/conding-challenge-i/src/App.tsx

```diff
@@ -0,0 +1,94 @@
+import { useState } from 'react'
+import './App.css'
+
+import CardName from './components/CardName'
+
+const randomInt = (min:number, max:number) => Math.floor(Math.random() * (max - min + 1)) + min;
+
+function findClosest(arr:number[],  target:number)
+{
+  let n = arr.length;
+
+  if (target <= arr[0])
+    return 0;
+
+  if (target >= arr[n-1])
+    return n-1;
+
+  let i = 0, j = n, mid= 0;
+
+  while(i<j)
+  {
+
+    mid = Math.floor((i+j) / 2);
+
+    if (arr[mid] == target)
+      return mid;
+
+    if (target < arr[mid]){
+      if (mid>0 && target>arr[mid-1])
+        return getClosest(mid - 1,
+                          mid, target, arr);
+      j = mid
+    }
+    else
+    {
+      if (mid < n - 1 && target < arr[mid+1])
+        return getClosest(mid, mid+1, target, arr);
+      i = mid + 1;
+    }
+  }
+
+  return mid;
+}
+
+function getClosest(curr_element:number,
+                    next_element:number,
+                    target:number, arr:number[]){
+  let val1 = arr[curr_element]
+  let val2 = arr[next_element]
```

Comment:
Cuando los valores de una variable no van a cambiar despues de ser asignado su valor, la buena practica es declararlo con `const` 
```suggestion
  const val1 = arr[curr_element]
  const val2 = arr[next_element]
```